.azure-pipelines.yml

```diff
@@ -33,3 +40,19 @@ jobs:
       - template: .ci/wasm-pack-test-headless.yml
         parameters:
           browser_flag: --chrome
+  - job: build_book
+    condition: |
+      eq(variables['Build.Reason'], 'PullRequest')
+    displayName: "Doc - build the book"
+    steps:
+      - template: .ci/install-mdbook.yml
+      - script: (cd guide && mdbook build)
+  - job: deploy_book
+    displayName: "Doc - deploy the book"
+    condition: |
+      and(not(eq(variables['Build.Reason'], 'PullRequest')),
+          eq(variables['Build.SourceBranch'], 'refs/heads/master'))
```

Comment:
You should only need this line and I belive you can get rid of all `variables['Build.Reason'], 'PullRequest')`

Reply:
This condition can from [this doc](https://cloudblogs.microsoft.com/opensource/2019/04/05/publishing-github-pages-from-azure-pipelines/?WT.mc_id=twitter-social-rylevick):

> That condition is a little wild as well. You can read it like a prefix-notation functional language (or an Excel formula, if you prefer): “Run this step only if the variable Build.Reason is NOT ‘PullRequest’ and the variable Build.SourceBranch is ‘master’.”